src/miner.cpp

```diff
@@ -52,95 +52,97 @@ using namespace zen;
 // BitcoinMiner
 //
 
-//
-// Unconfirmed transactions in the memory pool often depend on other
-// transactions in the memory pool. When we select transactions from the
-// pool, we select by highest priority or fee rate, so we might consider
-// transactions that depend on transactions that aren't yet in the block.
-// The COrphan class keeps track of these 'temporary orphans' while
-// CreateBlock is figuring out which transactions to include.
-//
-class COrphan
-{
-public:
-    const CTransactionBase* ptx;
-    set<uint256> setDependsOn;
-    CFeeRate feeRate;
-    double dPriority;
-
-    COrphan(const CTransactionBase* ptxIn) : ptx(ptxIn), feeRate(0), dPriority(0)
-    {
-    }
-};
-
 uint64_t nLastBlockTx = 0;
 uint64_t nLastBlockSize = 0;
 
-// We want to sort transactions by priority and fee rate, so:
-class TxPriorityCompare
+bool TxPriorityCompare::operator()(const TxPriority& a, const TxPriority& b)
 {
-    bool byFee;
-
-public:
-    TxPriorityCompare(bool _byFee) : byFee(_byFee) { }
-
-    bool operator()(const TxPriority& a, const TxPriority& b)
+    // first of all, if we are comparing two certs, we must be sure they are ordered by
+    // quality if they refer to the same scid and epoch; this criterion is a consensus rule
+    // and overrides the others two
+    if (a.get<2>()->IsCertificate() && b.get<2>()->IsCertificate() )
     {
-        // first of all, if we are comparing two certs, we must be sure they are ordered by
-        // quality if they refer to the same scid and epoch; this criterion is a consensus rule
-        // and overrides the others two
-        if (a.get<2>()->IsCertificate() && b.get<2>()->IsCertificate() )
-        {
-            // dynamic casting throws an exception upon failure
-            try {
-                const CScCertificate& aCert = dynamic_cast<const CScCertificate&>(*a.get<2>());
-                const CScCertificate& bCert = dynamic_cast<const CScCertificate&>(*b.get<2>());
- 
-                if (aCert.GetScId() == bCert.GetScId() )
-                {
-                    LogPrint("sc", "%s():%d - qa=%d - qb=%d\n", __func__, __LINE__, aCert.quality, bCert.quality );
-
-                    if (aCert.epochNumber != bCert.epochNumber)
-                        assert("two certs for the same scid can not have different epochs" == 0);
```

Comment:
The check on epoch is not performed in the modified version; we can probably live without it since that should already have been performed in mempool

Reply:
I believe that check is not essential to the right ordering of certs in block. I would leave it to the mempool as you mentioned